### PR TITLE
AG-7867 Number formatting rework

### PR DESCRIPTION
**Pull request by alexanderby**

### CODE REVIEW

File: charts-packages/ag-charts-community/src/util/numberFormat.ts [hunk 1]
```diff
@@ -1,672 +1,301 @@
...
+                  result = result.padEnd(w, f);
+              } else if (align === '^') {
+                  const padWidth = Math.max(0, w - result.length);
+                  const padLeft = Math.ceil(padWidth / 2);
+                  const padRight = Math.floor(padWidth / 2);
+                  result = result.padStart(padLeft + result.length, f);
+                  result = result.padEnd(padRight + result.length, f);
+              }
+              return result;
+          };
```
Comment (alantreadway) [inline charts-packages/ag-charts-community/src/util/numberFormat.ts:143]: Nit: Maybe factor out a `pad()` function to simplify the enclosing function?

File: charts-packages/ag-charts-community/src/util/numberFormat.ts [hunk 2]
```diff
@@ -1,672 +1,301 @@
...
+siPrefixes.set(-3, 'm');
+siPrefixes.set(0, '');
+siPrefixes.set(3, 'k');
+siPrefixes.set(6, 'M');
+siPrefixes.set(9, 'G');
+siPrefixes.set(12, 'T');
+siPrefixes.set(15, 'P');
+siPrefixes.set(18, 'E');
+siPrefixes.set(21, 'Z');
+siPrefixes.set(maxSIPrefix, 'Y');
```
Comment (alantreadway) [inline charts-packages/ag-charts-community/src/util/numberFormat.ts:275]: Could we just use a regular object here?

**Review by alantreadway — COMMENTED** [review]
Very nice work, just a couple of minor points!

File: charts-packages/ag-charts-community/src/util/numberFormat.ts (hunk 2 above)
Comment (alexanderby) [reply charts-packages/ag-charts-community/src/util/numberFormat.ts:275]: Personally I don't like objects having fake numeric keys:
```ts

const siPrefixes: Record<number, string> = {
    [minSIPrefix]: 'y',
    '-21': 'z',
    '-18': 'a',
    '-15': 'f',
    '-12': 'p',
    '-9': 'n',
    '-6': 'µ',
    '-3': 'm',
    0: '',
    3: 'k',
    6: 'M',
    9: 'G',
    12: 'T',
    15: 'P',
    18: 'E',
    21: 'Z',
    [maxSIPrefix]: 'Y',
};
```
But there is a way to make map look a bit better:
```ts
const siPrefixes = new Map<number, string>()
    .set(minSIPrefix, 'y')
    .set(-21, 'z')
    .set(-18, 'a')
    .set(-15, 'f')
    .set(-12, 'p')
    .set(-9, 'n')
    .set(-6, 'µ')
    .set(-3, 'm')
    .set(0, '')
    .set(3, 'k')
    .set(6, 'M')
    .set(9, 'G')
    .set(12, 'T')
    .set(15, 'P')
    .set(18, 'E')
    .set(21, 'Z')
    .set(maxSIPrefix, 'Y');
```

Comment (alantreadway) [reply charts-packages/ag-charts-community/src/util/numberFormat.ts]: How about:
```
const siPrefixes: Record<number, string> = {
    [minSIPrefix]: 'y',
    [-21]: 'z',
    [-18]: 'a',
    [-15]: 'f',
    [-12]: 'p',
    [-9]: 'n',
    [-6]: 'µ',
    [-3]: 'm',
    [0]: '',
    [3]: 'k',
    [6]: 'M',
    [9]: 'G',
    [12]: 'T',
    [15]: 'P',
    [18]: 'E',
    [21]: 'Z',
    [maxSIPrefix]: 'Y',
};
```

The use of `Map` for a simple lookup structure looks suspicious (as-in, exceptional), which hurts readability a little IMO.

**Review by alantreadway — APPROVED** [review]
LGTM - thanks for the updates!

**Pull request merged**